source/app/util/custom_resource.py

```diff
@@ -145,6 +145,9 @@ def handle_request(self):
         timer = threading.Timer(timeleft, self.fn_timeout)
         timer.start()
 
+        # default error to return if something causes the try to fail (this has been freezing for some reason)
+        self.response["Status"] = "FAILED"
+        self.response["Reason"] = "Unknown Error"
```

Comment:
Bug: on completing a successful request, your response object contains the field `Reason` with the value `Unknown Error`.

Reply:
yeah good catch. is this change even worth pushing through at all right now? My goal was to try to add more ways for it to fail rather than failing silently (I was not able to find an error anywhere in the logs). I don't have enough information to know what caused the failure or how to make it repeatable/testable. So I see 3 options:
1 - do nothing, deal with it later
2 - refactor this custom resource 'properly' and add tests around the new design
3 - patch in some extra error reporting but otherwise leave the behavior alone (this is what I aimed for, but may have failed to achieve)